### ISSUE TITLE
A packed object with `$graph` can be loaded via `load_document_by_yaml` but the loaded object cannot be saved as the packed object

### ISSUE THREAD

**Issue by tom-tan** [common-workflow-language/cwl-utils#98]
We usually expect the loaded CWL object can be saved as a valid CWL document but currently we cannot do that with packed objects with `$graph`.

To solve this issue, we have to fix the [`save`](https://github.com/common-workflow-language/cwl-utils/blob/3ec1a86745b552b242b8d9ae9cc58cde2e3f0828/cwl_utils/parser_v1_2.py#L123) function that is generated by [schema-salad-tool](https://github.com/common-workflow-language/schema_salad), I guess.

The problem is about the `cwlVersion` field in the root level.
We cannot fix `save` by adding the hardcoded `cwlVersion` field to the resulted dictionary because the top level `cwlVersion` is CWL-specific structure and `schema-salad-tool --codegen` should not assume the structure of CWL that are not given by the [schema](https://github.com/common-workflow-language/common-workflow-language/blob/1c1f122f780075d910fdfdea7e15e46eef3c078d/v1.0/CommonWorkflowLanguage.yml).

Do you have any idea for that?

# How to reproduce
Here is a code (`test.py`) to load and save a given CWL document. It is based on the [example in README](https://github.com/common-workflow-language/cwl-utils#using-the-cwl-parsers).
```python
#!/usr/bin/env python
# Imports
from pathlib import Path
from ruamel import yaml
import sys

# File Input - This is the only thing you will need to adjust or take in as an input to your function:
cwl_file = Path(sys.argv[1]).resolve(strict=True) # modified from the example

# Read in the cwl file from a yaml
with open(cwl_file, "r") as cwl_h:
 yaml_obj = yaml.main.round_trip_load(cwl_h, preserve_quotes=True)
    
# Check CWLVersion
if 'cwlVersion' not in list(yaml_obj.keys()):
    print("Error - could not get the cwlVersion")
    sys.exit(1)

# Import parser based on CWL Version    
if yaml_obj['cwlVersion'] == 'v1.0':
    from cwl_utils import parser_v1_0 as parser
elif yaml_obj['cwlVersion'] == 'v1.1':
    from cwl_utils import parser_v1_1 as parser
elif yaml_obj['cwlVersion'] == 'v1.2':
    from cwl_utils import parser_v1_2 as parser
else:
    print("Version error. Did not recognise {} as a CWL version".format(yaml_obj["cwlVersion"]))
    sys.exit(1)

# Import CWL Object
cwl_obj = parser.load_document_by_yaml(yaml_obj, cwl_file.as_uri())

# Export CWL Object (added)
import json
print(json.dumps(parser.save(cwl_obj)))
```

Input: [scatter-wf3.cwl](https://github.com/common-workflow-language/common-workflow-language/blob/1c1f122f780075d910fdfdea7e15e46eef3c078d/v1.0/v1.0/scatter-wf3.cwl) (Note: it is a [packed object](https://www.commonwl.org/v1.2/CommandLineTool.html#Packed_documents) with `$graph` and `cwlVersion` field)

```console
$ ./test.py scatter-wf3.cwl | jq '.["cwlVersion"]'
```

# Expected behavior
```console
$ ./test.py scatter-wf3.cwl | jq '.["cwlVersion"]'
"v1.0"
```

# Actual behavior
```console
$ ./test.py scatter-wf3.cwl | jq '.["cwlVersion"]'
jq: error (at <stdin>:1): Cannot index array with string "cwlVersion"
```

**Comment by mr-c** [comment]
I think this repo can container wrapper code in a separate file for this and similar situations. Not all code has to be from `schema-salad-tool --codegen`

**Comment by tom-tan** [comment]
Fixed by #99!
